community/kernel/src/main/java/org/neo4j/kernel/impl/api/TransactionToApply.java

```diff
@@ -123,11 +126,20 @@ public void commitment( Commitment commitment, long transactionId )
     {
         this.commitment = commitment;
         this.transactionId = transactionId;
+        if ( callback != null )
+        {
+            commitment.onClosed( callback );
+        }
     }
 
     @Override
     public TransactionToApply next()
     {
         return nextTransactionInBatch;
     }
+
+    public void onClose( Consumer<Long> callback )
```

Comment:
I don't think the correct term is used here. It's called `onClose`, but looking at where it's called (from `committment(...)`) it's more of a `onCommitted`, i.e. before being applied. Where in the life cycle of a transaction do you intend to hook this in?